### ISSUE TITLE
PythonParser: should parse Pipfiles

### ISSUE THREAD

**Issue by nokome** [stencila/dockta#4]
The `PythonParser` should be able to parse Pipfiles https://github.com/pypa/pipfile

**Comment by nokome** [comment]
... or [`Pipfile.lock`](https://pipenv.readthedocs.io/en/latest/basics/#example-pipfile-lock) if it exists

**Comment by giorgiosironi** [comment]
Just to clarify, what is the purpose of parsing the file? It could be treated as a black box for the purposes of just installing the dependencies, as it can be passed directly to `pipenv`.

**Comment by nokome** [comment]
@giorgiosironi : yes, that's a good question. The only reason that we do it is so we can generate a meta-data tree (`environ.jsonld`) containing with information on dependencies. From that you could generate a software citation list for your project or publish a more structured, semantic, and language agnostic representation of your project and it's dependencies.

**Comment by remram44** [comment]
Also consider supporting `pyproject.toml` (or `poetry.lock`), see [PEP 518](https://www.python.org/dev/peps/pep-0518/) and [Poetry](https://python-poetry.org/docs/pyproject/)

**Comment by nokome** [comment]
Thanks @remram44 👋. Yep agreed, that would be good to add.
